client.go

```diff
@@ -330,5 +287,16 @@ func NewAuthenticatedClient(versionedURL, apiKey string) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &Client{Signer: signer, APIURL: parsedURL}, nil
+
+	httpClient := retryablehttp.NewClient()
+
+	// Need to re-sign the request before each retry
+	httpClient.RequestLogHook = func(logger retryablehttp.Logger, request *http.Request, count int) {
```

Comment:
We'd need to adapt a github.com/juju/loggo logger as the log writer implementation